src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/ConfigurationMockTests.cs

```diff
@@ -31,228 +29,371 @@ public class ConfigurationMockTests
             }
         };
 
-        private static (ConfigurationClient service, TestPool<byte> pool) CreateTestService(MockHttpClientTransport transport)
+        private static ConfigurationClient CreateTestService(HttpPipelineTransport transport)
         {
             HttpPipelineOptions options = ConfigurationClient.CreateDefaultPipelineOptions();
-            var testPool = new TestPool<byte>();
-            options.AddService(testPool, typeof(ArrayPool<byte>));
-
             options.Transport = transport;
-
-            var service = new ConfigurationClient(connectionString, options);
-
-            return (service, testPool);
-        }
-
-        private static bool TagsEqual(IDictionary<string, string> expected, IDictionary<string, string> actual)
-        {
-            if (expected == null && actual == null) return true;
-            if (expected?.Count != actual?.Count) return false;
-            foreach (var pair in expected)
-            {
-                if (!actual.TryGetValue(pair.Key, out string value)) return false;
-                if (!string.Equals(value, pair.Value, StringComparison.Ordinal)) return false;
-            }
-            return true;
+            return new ConfigurationClient(connectionString, options);
         }
 
         [Test]
         public async Task Get()
         {
-            var transport = new GetMockTransport(s_testSetting.Key, default, s_testSetting);
-            var (service, pool) = CreateTestService(transport);
+            var response = new MockResponse(200);
+            response.SetContent(SerializationHelpers.Serialize(s_testSetting, SerializeSetting));
+
+            var mockTransport = new MockTransport(response);
+            ConfigurationClient service = CreateTestService(mockTransport);
 
             ConfigurationSetting setting = await service.GetAsync(s_testSetting.Key);
 
+            MockRequest request = mockTransport.SingleRequest;
+
+            AssertRequestCommon(request);
+            Assert.AreEqual(HttpPipelineMethod.Get, request.Method);
+            Assert.AreEqual("https://contoso.azconfig.io/kv/test_key", request.Uri.ToString());
```

Comment:
Looking at Get and GetLabel, the implementation is the same, the difference is the Uri. 
How are you managing to avoid the `?label=test_label` here if the object that we are passing is the same `s_testSetting`?

Reply:
GetLabel passes second parameter into GetAsync: https://github.com/Azure/azure-sdk-for-net/pull/5708/files/db5f84ef26dd0ee986d5b4fe0450129fa2bb63b8#diff-ceba0fdb8b4f0470e5a6eb1f7fed2201R68

Reply:
Interesting... it this case, it works because it is a mock. 
If this was a live test, the answer from the server will be 404. Not found, because the setting that was saved in the Configuration Store does contain a label.

Should we reflect that too here in the mock? or because we are testing that we can do a get and how the content is sent, it doesn't matter...

Reply:
I would say the later. We are not trying to simulate the server just asserting that the client generates the right request when given the inputs.

Reply:
Yeah, sounds good to me